crates/xline/src/conflict/mod.rs

```diff
@@ -0,0 +1,51 @@
+use curp::cmd::Command as CurpCommand;
+use utils::interval_map::Interval;
+use xlineapi::{command::Command, interval::BytesAffine, RequestBackend, RequestWrapper};
+
+/// Speculative pool implementations
+pub(crate) mod spec_pool;
+/// Uncommitted pool implementations
+pub(crate) mod uncommitted_pool;
+
+/// Returns command intervals
+fn intervals<C>(entry: &C) -> Vec<Interval<BytesAffine>>
```

Comment:
`intervals` generate the intervals representation of the keys of a given command, it only needs to take a reference of the keys. `filter_kv` used to filter out the kv commands so it takes `C` and return `Option<C>` to avoid cloning.